### PR TITLE
fix: ReferenceError: onScriptComplete is not defined when using HMR o…

### PR DESCRIPTION
**Pull request by chris-czopp**
…n Firefox 45

**What kind of change does this PR introduce?**

bugfix for Firefox 45 and code improvement

**Did you add tests for your changes?**

no, it's an enhancement covered by existing tests

**If relevant, link to documentation update:**

N/A

**Summary**

I've been testing my app on out-dated Firefox (45) and realised the template relies on hoisting (`onScriptComplete` assigned before its declaration) which is not only a bad practice but it was crashing the bundling.

**Does this PR introduce a breaking change?**

no, this is just enhancement

### CODE REVIEW

**Comment by aleen42** [comment]
Met the same problem when using Firefox 40, but I have tested that it did not seem the problem of hoisted, as the following snippet has worked under Firefox 40:

```js
(function () {
    console.log(test);
    function test() {}
})();
```

**Comment by evilebottnawi** [comment]
@chris-czopp Please accept CLA

**Comment by chris-czopp** [comment]
@aleen42 I rushed too quickly with the conclusions, good to know the root cause of the issue :)
@evilebottnawi I did and then realised my email address in gitconfig is wrong, I changed it and now I'm stuck with the signed CLA and GH check not being able to pick this up...

**Comment by evilebottnawi** [comment]
@chris-czopp your should rewrite your email in commints

**Comment by aleen42** [comment]
@sokra method in the timeout handler function before has the same problem.

**Comment by evilebottnawi** [comment]
@aleen42 will be merged in near future
/cc @sokra

File: lib/web/JsonpMainTemplatePlugin.js [hunk 1]
```diff
@@ -173,7 +173,6 @@ class JsonpMainTemplatePlugin {
 						"onScriptComplete({ type: 'timeout', target: script });"
```
[the remark this answers is not in the document]
Comment (chris-czopp) [reply lib/web/JsonpMainTemplatePlugin.js]: yes, however the `timeout` var is used inside `onScriptComplete`. If we moved it below we'd have `timeout` var hoisted... I think it needs to be further refactored.

**Comment by sokra** [comment]
Thanks, more changes can be done in a separate PR

**Pull request merged**